spec/smart_app_launch/well_known_capabilities_test_spec.rb

```diff
@@ -0,0 +1,142 @@
+require_relative '../../lib/smart_app_launch/well_known_capabilities_stu1_test'
+require_relative '../../lib/smart_app_launch/well_known_capabilities_stu2_test'
+
+RSpec.describe "Well-Known Tests" do
+  let(:test_v1) { Inferno::Repositories::Tests.new.find('well_known_capabilities_stu1') }
+  let(:test_v2) { Inferno::Repositories::Tests.new.find('well_known_capabilities_stu2') }
+  let(:session_data_repo) { Inferno::Repositories::SessionData.new }
+  let(:test_session) { repo_create(:test_session, test_suite_id: 'smart') }
+  let(:well_known_config) do
+    {
+      'authorization_endpoint' => 'https://example.com/fhir/auth/authorize',
+      'token_endpoint' => 'https://example.com/fhir/auth/token',
+      'token_endpoint_auth_methods_supported' => ['client_secret_basic'],
+      'registration_endpoint' => 'https://example.com/fhir/auth/register',
+      'scopes_supported' =>
+        ['openid', 'profile', 'launch', 'launch/patient', 'patient/*.*', 'user/*.*', 'offline_access'],
+      'response_types_supported' => ['code', 'code id_token', 'id_token', 'refresh_token'],
+      'management_endpoint' => 'https://example.com/fhir/user/manage',
+      'introspection_endpoint' => 'https://example.com/fhir/user/introspect',
+      'revocation_endpoint' => 'https://example.com/fhir/user/revoke',
+      'capabilities' =>
+        ['launch-ehr', 'client-public', 'client-confidential-symmetric', 'context-ehr-patient', 'sso-openid-connect'],
+      'issuer' => 'https://example.com',
+      'jwks_uri' => 'https://example.com/.well-known/jwks.json',
+      'grant_types_supported' => ['authorization_code'],
+      'code_challenge_methods_supported' => ['S256']
+    }
+  end
+
+  def run(runnable, inputs = {})
+    test_run_params = { test_session_id: test_session.id }.merge(runnable.reference_hash)
+    test_run = Inferno::Repositories::TestRuns.new.create(test_run_params)
+    inputs.each do |name, value|
+      session_data_repo.save(
+        test_session_id: test_session.id,
+        name: name,
+        value: value,
+        type: runnable.config.input_type(name)
+      )
+    end
+    Inferno::TestRunner.new(test_session: test_session, test_run: test_run).run(runnable)
+  end
+
+  shared_examples 'well-known tests' do
```

Comment:
This is cool -- I didn't know RSpec had this. Also, the `valid_config.dup` removals below don't work in the shared examples, maybe shared examples doesn't give a fresh let binding to each test? Either way not worth the hassle of actually digging into it, so left those as-is.

Reply:
i.e.

let is defined in the `it_behaves_like` here:

https://github.com/inferno-framework/smart-app-launch-test-kit/blob/0e4c0dd7d256840b24558b8164ea350f1bb4e9fa/spec/smart_app_launch/well_known_capabilities_test_spec.rb#L101-L105

but outside it here:

https://github.com/inferno-framework/smart-app-launch-test-kit/blob/0e4c0dd7d256840b24558b8164ea350f1bb4e9fa/spec/smart_app_launch/well_known_capabilities_test_spec.rb#L109-L113

In the second case we needed to define it outside because there are additional tests that use it.